[easy][indexer] add README (#6978)
MystenLabs/sui: crates/sui-indexer/README.md

Sui indexer is an off-fullnode service to serve data from Sui protocol, including both data directly generated from chain and derivative data.

## Current architecture Dec 2022 (will change soon)
![indexer_simple](https://user-images.githubusercontent.com/106119108/209000367-4c7d23d8-fef2-4485-8472-89c31f0e2d62.png)

## Steps to run locally
### Prerequisites
- install local [Postgres server](https://www.postgresql.org/download/)
- install Diesel CLI, you can follow the [Diesel Getting Started guide](https://diesel.rs/guides/getting-started) up to the *Write Rust* section
- [optional but handy] Postgres client like [Postico](https://eggerapps.at/postico2/), for local check, query execution etc.

### Steps
1. DB setup
```sh
# DB setup, run the following commands from the /sui-indexer folder
# .env file under /sui-indexer is required for diesel cmds
# in .env file, DATABASE_URL should point to your local PG server
# an example is:

> # DATABASE_URL=""postgres://postgres:postgres@localhost/gegao"

Reviewer: ```suggestion
# DATABASE_URL="postgres://postgres:postgres@localhost/gegao"
```
nit

diesel setup

# and then run 
diesel migration run
```
2. cargo run under `/sui-indexer`
```sh

> cargo run --bin sui-indexer -- --db-url "<LOCAL_DB_URL>" --rpc-client-url "https://fullnode.devnet.sui.io:443"

Reviewer: Is `LOCAL_DB_URL` same as `postgres://postgres:postgres@localhost/gegao` above?

Author: Yes, will change it to DATABASE_URL for consistency then.

Sui indexer is an off-fullnode service to serve data from the protocol, including both data directly generated from chain and derivative data.

> ## today's architecture (will be changed soon)

Reviewer 2: ```suggestion
## Current architecture Dec 2022 (will change soon)
```

![indexer_simple](https://user-images.githubusercontent.com/106119108/209000367-4c7d23d8-fef2-4485-8472-89c31f0e2d62.png)

> ## local run instructions

Reviewer 2: ```suggestion
## Steps to run locally
```

### Prerequisites
- install local [Postgres server](https://www.postgresql.org/download/)

> - install Diesel CLI, you can follow this [guide](https://diesel.rs/guides/getting-started) until “write Rust” section

Reviewer 2: ```suggestion
- install Diesel CLI, you can follow the [Diesel Getting Started guide](https://diesel.rs/guides/getting-started) up to the *Write Rust* section
```

- [optional but handy] Postgres client like [Postico](https://eggerapps.at/postico2/), for local check, query execution etc.

### Steps
1. DB setup
  ```

> // DB setup, run below under /sui-indexer

Reviewer 2: ```suggestion
// DB setup, run the following commands from the /sui-indexer folder
```

// .env file under /sui-indexer is required for diesel cmds
// in .env file, DATABASE_URL should point to your local PG server
// an example is:
// DATABASE_URL=""postgres://postgres:postgres@localhost/gegao"

> diesel setup
> diesel migration run

Reviewer 2: ```suggestion
`diesel setup`

and then run 

`diesel migration run`
```
I couldn't suggest an update that uses ```sh as it did not render in the comment

  ```
2. cargo run under `/sui-indexer`

>   ```

Reviewer 2: ```suggestion
  ```sh
```

  cargo run --bin sui-indexer -- --db-url "<LOCAL_DB_URL>" --rpc-client-url "https://fullnode.devnet.sui.io:443"
  ```
  
### Clean up and re-run

>   ```

Reviewer 2: might be more readable if this was just text and then the command could be code format instead the whole section being a code-formatted block

Author: good idea and changed accordingly!

>   // clean up all tables in the local DB, run below until no more tables are deleted

Reviewer 2: ```suggestion
  // clean up all tables in the local DB, run the following command until no more tables are deleted
```

>   diesel migration revert

Reviewer 2: ```suggestion
`diesel migration revert`
```
